app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java

```diff
@@ -44,10 +54,35 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.menu_save_search_engine:
-                // TODO: Save engineName and searchString
+                final View rootView = getView();
+                final String engineName = ((EditText) rootView.findViewById(R.id.edit_engine_name)).getText().toString();
+                final String searchQuery = ((EditText) rootView.findViewById(R.id.edit_search_string)).getText().toString();
+
+                final SharedPreferences sharedPreferences = getActivity()
+                        .getSharedPreferences(SearchEngine.PREF_FILE_SEARCH_ENGINES, Context.MODE_PRIVATE);
+                if (!validateSearchFields(engineName, searchQuery, sharedPreferences)) {
+                    Snackbar.make(rootView, R.string.search_add_error, Snackbar.LENGTH_SHORT).show();
+                } else {
+                    SearchEngine.addSearchEngine(sharedPreferences, getActivity(), engineName, searchQuery);
+                    Snackbar.make(rootView, R.string.search_add_confirmation, Snackbar.LENGTH_SHORT).show();
+                    getActivity().onBackPressed();
```

Comment:
Should this be finish() or fragmentManager.popBackStack() ? Or does onBackPressed() do something additionally?

Reply:
onBackPressed does the combination of popBackStack and finish - for this case, I guess it doesn't really matter because it's not the last view, but maybe it makes more sense for the Fragment to use the FragmentManager call.